backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -114,29 +107,65 @@ private MatchPreference getCombinedMatchPreference(
    */
   private boolean isCombinedMatchPreferenceSatisfied(
       MatchPreference combinedMatchPreference,
-      Participant newParticipant,
-      Participant currParticipant) {
+      Participant firstParticipant,
+      Participant secondParticipant) {
     if (combinedMatchPreference == MatchPreference.ANY) {
+      // both ANY, doesn't matter how many same fields
       return true;
     }
 
-    // Count number of same fields between the two participants
-    boolean sameRole = newParticipant.getRole().equals(currParticipant.getRole());
-    boolean sameProductArea =
-        newParticipant.getProductArea().equals(currParticipant.getProductArea());
-    int numSameFields = Booleans.countTrue(sameRole, sameProductArea);
-    System.out.println("number of same fields: " + numSameFields);
+    // Get lists of combined role, product area, interests for first and second participant
+    String firstFilledFieldsList = getAllFilledFieldsList(firstParticipant);
```

Comment:
What do you think about renaming this to firstParticipantInputs or something similar? Also similar comment about the method name, would it be possible to rename it to something without "List" at the end since it's returning a String ( maybe something like getAllParticipantInputs)?